model/relabel/relabel.go

```diff
@@ -188,10 +188,14 @@ func (re Regexp) String() string {
 	return str[4 : len(str)-2]
 }
 
-// Process returns a relabeled copy of the given label set. The relabel configurations
+// Process returns a relabeled version of the given label set. The relabel configurations
 // are applied in order of input.
+// If any relabel configuration has a label from the input label as target label the input
+// label set will be modified.
+// There may be other circumstances where Process may modify the input label.
+// If you want to avoid issues with the input label set being modified, at the cost of
+// higher memory usage, you can use labels.Labels.Copy().
 // If a label set is dropped, nil is returned.
```

Comment:
With these changes to the comment any user of this function will know:

- At least one scenario where a modification of the input happens (L193-194).
- How to avoid having issues with this and its consequences (L196-197).

This should help people avoid issues if in the future more cases of input modification are implemented.